storage/statedb/database.go

```diff
@@ -822,9 +829,22 @@ func (db *Database) Commit(node common.Hash, report bool) error {
 	db.gcnodes, db.gcsize, db.gctime = 0, 0, 0
 	db.flushnodes, db.flushsize, db.flushtime = 0, 0, 0
 
+	if db.settingMigrationDBRequired(blockNum) {
+		db.diskDB.SetStateTrieMigrationDB(blockNum)
+	}
 	return nil
 }
 
+func (db *Database) settingMigrationDBRequired(blockNum uint64) bool {
+	if blockNum == NoDataArchivingPreparation {
+		return false
+	}
+	if blockNum >= db.dataArchivingBlockNum && !db.diskDB.InMigration() {
```

Comment:
Shouldn't it work if DataArchivingBlockNum is 0?
 because of comment `DataArchivingBlockNum uint64 // If not zero, the number indicates the starting point of data archiving operation.`

Reply:
@nogang You're right. I missed that point. Please check the last commit =) Thanks!